src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
```

Comment:
Just an extra thought here: since you're already using array functions here maybe you should use `.map()` and instead of pushing into and external array for each element in action, simply return the resulting state

Reply:
great heads up @EdPirro!!!!